### ISSUE TITLE
AGPL license is not used properly

### ISSUE THREAD

**Issue by ian-kelling** [thecodingmachine/workadventure#678]
https://www.gnu.org/licenses/gpl-faq.en.html#ModifyGPL

Yours does the things that link says not to.

**Comment by ian-kelling** [comment]
To be specific, some of the problems: 

The license text in your source does follow https://www.gnu.org/licenses/gpl-faq.en.html#ModifyGPL. 

https://workadventu.re/faq says "The self-hosted version of WorkAdventure is distributed under the AGPL v3 license modified by the Commons Clause", does not fit within the spirit of https://www.gnu.org/licenses/gpl-faq.en.html#ModifyGPL text which says "call your license by another name."

**Comment by timmwille** [comment]
True, so it must be their own license. Sad, because I think if you stay ahead with the development you will also be the one to go to so the monetizing argument is not really the main point here. So creating a Freemium License out of the AGPL but essentially giving it a touch of a CC NC license but appreciating the Open Source community sounds really a bit unsafe for collaborators. 

We now have to discuss if we can even use it in our non-profit organisation. 

Basically the already quite strong AGPL modification says: "keep it open if you run/adapt it, but do not use it the way it is opened"

It is a bit like the CC-BY-NC-SA now, which is a huge [discussion](https://mifactori.de/non-commercial-is-not-open-source/) in the Open Source Community (Hardware is the field I know best). I do think this software has great potential, especially when opened up for the Open Source Community to collaborate on. 

So the Commons Clause is actually right now the Opposite of "Commons" → I'm pretty sure the intention is good and I can to some extent even understand why the addition was made. But I do think from all I know about those (especially copyleft) licenses, this adoption creates more trouble then it tries to solve (security for collaborators, non-profit use, educational context, compatibility with other software/licenses, conflict as mentioned in this issue above and more legal issues/costs if it will actually be used as SaaS and it goes to a legal case). 

My Proposal: keep at least an open core and/or add a Code-of-Conduct.md 
https://github.com/thecodingmachine/workadventure/community/code-of-conduct/new?template=contributor-covenant


With all the best intentions to help, cheers and keep up the good work! :heart_decoration:

**Comment by timmwille** [comment]
Ah sorry it is already in discussion with @mariobehling in here:
https://github.com/thecodingmachine/workadventure/issues/678

**Comment by timmwille** [comment]
so in other words this might be ok to close and we continue in the other issue @moufmouf

**Comment by timmwille** [comment]
by the way, for the lawyers around you, from my understanding every contributor provides his/her bits of the project under the same conditions as the forked repo contains right? so meaning in the end all those nice changes can not be used by anyone as SaaS, as it is a mix of many contributors who forbid eachother's use of their bits of the puzzle? I really think the Commons clause is creating quite a drama as a license and can maybe be put in a code of conduct at least?

**Comment by valerio-bozzolan** [comment]
BTW

A couple of months ago [Italian Linux Society](https://www.ils.org/) and Wikimedia DE contacted @moufmouf to discuss about this issue. [Here](https://meta.wikimedia.org/wiki/WikiProject_remote_event_participation/WorkAdventure) a report of that discussion.

At the end Italian Linux Society requested a price quotation to Free the frontend (removing Google Analytics and replacing it with Matomo or something like that, and bonus point: [LibreJS](https://www.gnu.org/software/librejs/) support :1st_place_medal: ) to then discuss in a next future about Freeing the backend as well (the AGPL issue).

Sadly I have not received any response since weeks and I am somewhat worried that my emails will end up in SPAM. It happens sometime. Can an internal confirm me that my emails (`direttore at linux dot it`) are received?

Thank you so much.

**Comment by moufmouf** [comment]
Hey @timmwille , hey @valerio-bozzolan ,

Thanks for your messages.
Sorry if we are leaving those issues unanswered for weeks. We have a ton of things to do here and are still a small team. We are putting focus on the features and bugfixes right now. I'll certainly address the licensing issue at some point of course, but for the coming weeks, we are focusing on mobile support, scripting and bug fixing.

> At the end Italian Linux Society requested a price quotation to Free the frontend (removing Google Analytics and replacing it with Matomo or something like that, and bonus point: LibreJS support 1st_place_medal ) to then discuss in a next future about Freeing the backend as well (the AGPL issue).

@valerio-bozzolan , we just wouldn't charge anything to simply remove Google Analytics. The current self-hosted version in Github does not have Google Analytics by default (you can enable it with an environment variable). And actually, you can use a special environment variable to enable any tracker (Matomo included).

Google Analytics is still used on our website (workadventu.re) and on the hosted game (play.workadventu.re). In the coming months, we plan to remove Google Analytics from the hosted game (play.workadventu.re) and replace it with home-grown analytics. We will keep using Google Analytics for the website in the long run (workadventu.re), but the website is only used by administrators, not the WorkAdventure players.

**Comment by valerio-bozzolan** [comment]
Thank you for the response!

Just to clarify my official position, I'm interested for a price quotation for:

- [ ] Solution 1: Making [play.workadventu.re](https://play.workadventu.re/) 100% Libre (just client-side is fine)
   - [ ] remove proprietary JS Google Analytics ← current problem
   - [ ] bonus point: LibreJS support 

Or:

- [ ] Solution 2: making the self-hosted version 100% Libre
    - [X] remove proprietary JS Google Analytics
    - [ ] remove proprietary server-side restrictions ("Common Clause") ← current problem
    - [ ] bonus point: LibreJS support 

Now. We would like to have (at least) one of these solutions. We are currently stalled. This is why I proposed to pay you to turbo the solution 1.

----

> home-grown analytics

BTW what's that?

Thank you for your clarifications.

**Comment by altsalt** [comment]
> > home-grown analytics
> 
> BTW what's that?
> 
> Thank you for your clarifications.

https://matomo.org/ ?

**Comment by TechnologyClassroom** [comment]
@valerio-bozzolan The FSF forked the last version before the commons clause was added and call it [LibreAdventure](https://vcs.fsf.org/?p=libreadventure.git;a=summary) to use with LibrePlanet.  I hope the commons clause is removed in the future so this won't be necessary.

**Comment by defnull** [comment]
@TechnologyClassroom FSF might be interested in #2317
